SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialVirtualWorkerTranslator.cpp

```diff
@@ -11,31 +13,64 @@ DEFINE_LOG_CATEGORY(LogSpatialVirtualWorkerTranslator);
 SpatialVirtualWorkerTranslator::SpatialVirtualWorkerTranslator()
 	: NetDriver(nullptr)
 {
+	bWorkerEntityQueryInFlight = false;
+	bIsReady = false;
+	DesiredVirtualWorkerCount = 0;
 }
 
 void SpatialVirtualWorkerTranslator::Init(USpatialNetDriver* InNetDriver)
 {
 	NetDriver = InNetDriver;
+	if (NetDriver != nullptr)
+	{
+		WorkerId = NetDriver->Connection->GetWorkerId();
+	}
+	else
+	{
+		WorkerId = "InvalidWorkerId";
+	}
 }
 
-const FString* SpatialVirtualWorkerTranslator::GetPhysicalWorkerForVirtualWorker(VirtualWorkerId id)
+void SpatialVirtualWorkerTranslator::SetDesiredVirtualWorkerCount(uint32 NumberOfVirtualWorkers)
 {
-	return VirtualToPhysicalWorkerMapping.Find(id);
-}
+	// Currently, this should only be called once on startup. In the future we may allow for more
+	// flexibility. 
+	if (bIsReady)
+	{
+		UE_LOG(LogSpatialVirtualWorkerTranslator, Warning, TEXT("(%s) SetDesiredVirtualWorkerCount called after the translator is ready, ignoring."), *WorkerId);
+	}
 
-void SpatialVirtualWorkerTranslator::ApplyVirtualWorkerManagerData(const Worker_ComponentData& Data)
-{
-	if (NetDriver)
+	DesiredVirtualWorkerCount = NumberOfVirtualWorkers;
```

Comment:
Is DesiredVirtualWorkerCount used anywhere?

(Ignore didn't realise it's a member)

That said, do we use it though?

Reply:
I think it will be used more in the future when/if we support changing the number of virtual workers, but for now, you're right, it's unnecessary. Removing it.